### DISCUSSION TITLE
Extracting outermost size of an array?

### DISCUSSION THREAD

**Discussion by MoAly98** [scikit-hep/awkward#1268] [Q&A]
Hi,
I ran into this issue when I simply wanted to check if an awkward array is empty or not after I filtered data from it. My array had a type that looks like:
```
N  * {"field1": X, "field2": var*Y}
```
which means that `ak.size` fails to compute the array size. I am wondering if there is a method to extract the  value `N` from the array without turning it into a list of lists? 

Thanks,
Mo

**Comment by agoose77** [comment]
`ak.size` is a function that implements `np.size` when the arrays are rectilinear. In your case, you do not have a rectilinear array, so this fails. The length `N` is just `len(array)`. For more complex length calculations there are `ak.count` and `ak.num`, which return slightly different length information that you can use.

**Reply by jpivarski** [reply]
That's right, [ak.count](https://awkward-array.readthedocs.io/en/latest/_auto/ak.count.html) and [ak.num](https://awkward-array.readthedocs.io/en/latest/_auto/ak.num.html) are the counting functions. You likely want [ak.num](https://awkward-array.readthedocs.io/en/latest/_auto/ak.num.html) for any "how many XYZs do I have?" questions, since this function treats records (`{"field1": X, "field2": var*Y}` objects) as units, whereas [ak.count](https://awkward-array.readthedocs.io/en/latest/_auto/ak.count.html) is a reducer that cuts across objects, as described more fully in the [ak.sum](https://awkward-array.readthedocs.io/en/latest/_auto/ak.sum.html) documentation.

Since your type is just

```
N  * {"field1": X, "field2": var*Y}
```

and I take it you want to know "`N`", just do `len(array)`. It happens to be the case that [ak.num](https://awkward-array.readthedocs.io/en/latest/_auto/ak.num.html) with `axis=0` is the same as `len(array)`.

----------------

Also, if you made the `array` with cuts by

```python
array = array_without_cuts[cuts]
```

then you can also do

```python
np.count_nonzero(cuts)
```

to get the number of entries that would pass without even applying the cut.
